lib/portage/tests/news/test_NewsItem.py

```diff
@@ -1,109 +1,347 @@
+
+        item = self._createNewsItem({"display_if_installed": ["sys-apps/portage"]})
+        self.assertTrue(item.isValid())
+        self.assertTrue(
+            item.isRelevant(self.vardb, self.settings, self.profile),
+            msg=f"Expected {item} to be relevant, but it was not!",
         )
 
-        try:
-            item = self._processItem(tmpItem)
-            self.assertTrue(
-                item.isRelevant(self.vardb, self.settings, self.profile),
-                msg=f"Expected {tmpItem} to be relevant, but it was not!",
-            )
-        finally:
-            os.unlink(item.path)
+        # Test the negative case: a single Display-If-Installed listing
+        # a package we don't have.
+        item = self._createNewsItem(
+            {"display_if_installed": ["sys-apps/i-do-not-exist"]}
+        )
+        self.assertTrue(item.isValid())
+        self.assertFalse(
+            item.isRelevant(self.vardb, self.settings, self.profile),
+            msg=f"Expected {item} to be irrelevant, but it was relevant!",
+        )
+
+        # What about several packages and we have none of them installed?
+        item = self._createNewsItem(
+            {
+                "display_if_installed": [
+                    "dev-util/pkgcheck",
+                    "dev-util/pkgdev",
+                    "sys-apps/pkgcore",
+                ]
+            }
+        )
+        self.assertTrue(item.isValid())
+        self.assertFalse(
+            item.isRelevant(self.vardb, self.settings, self.profile),
+            msg=f"Expected {item} to be irrelevant, but it was relevant!",
+        )
+
+        # What about several packages and we have one of them installed?
+        self.vardb.cpv_inject("net-misc/openssh-9.2_p1", {"SLOT": "0"})
+        item = self._createNewsItem(
+            {
+                "display_if_installed": [
+                    "net-misc/openssh",
+                    "net-misc/dropbear",
+                ]
+            }
+        )
+        self.assertTrue(item.isValid())
+        self.assertTrue(
+            item.isRelevant(self.vardb, self.settings, self.profile),
+            msg=f"Expected {item} to be relevant, but it was not!",
+        )
+
+        # What about several packages and we have all of them installed?
+        # Note: we already have openssh added from the above test
+        self.vardb.cpv_inject("net-misc/dropbear-2022.83", {"SLOT": "0"})
+        item = self._createNewsItem(
+            {
+                "display_if_installed": [
+                    "net-misc/openssh",
+                    "net-misc/dropbear",
+                ]
+            }
+        )
+        self.assertTrue(item.isValid())
+        self.assertTrue(
+            item.isRelevant(self.vardb, self.settings, self.profile),
+            msg=f"Expected {item} to be relevant, but it was not!",
+        )
+
+        # What if we have a newer version of the listed package?
```

Comment:
Is it maybe worth it to also check the case in which the required version is newer than our installed version?

Reply:
Sure, why not? Adding it in.